src/view/view.py

```diff
@@ -5,386 +5,316 @@
+        self.sudoku_checkbox_frame.checkboxes[0].select()
+        self.sudoku_checkbox_frame.checkboxes[1].configure(text="Edit Mode", command=self.set_edit_mode)
         self.sudoku_checkbox_frame.grid(row=2, column=1, padx=10, pady=10, sticky="nsew")
 
-        
-        self.tool_frame.grid_columnconfigure((0,1), weight=1)
-        # self.tool_frame.grid_rowconfigure((1,2), weight=0)
-        # self.tool_frame.grid_rowconfigure((0,3), weight=2)
-
+        self.tool_frame.grid_columnconfigure((0, 1), weight=1)
         self.tool_frame.grid(row=0, column=2, padx=10, pady=10, sticky="ew")
 
-
-
-
-
     def set_controller(self, controller):
         '''Sets the controller of the view'''
         self.controller = controller
+        self.bind_class("Entry", "<Button-1>", lambda *args: self.controller.push(), add="+")
 
-        self.bind_class("Entry","<Button-1>", lambda *args: self.controller.push(), add="+")
-
-
-        def on_keypress():
-            for row in range(9):
-                for column in range(9):
-                    self.invalid_field(row, column)
-        
-        self.bind_class("Entry", "<Button-1>", lambda args: on_keypress(), add="+")
-        
-        
-
-
-
-    def set_mouse_position (self, widget):
+    def set_mouse_position(self, widget):
         self.widget_at_mouse = widget
-        # print(f"Mouse position set to: {self.mouse_position}")
 
     def set_edit_mode(self):
         self.edit_mode = self.sudoku_checkbox_frame.checkboxes[1].get()
-        # print(f"Edit mode set to: {self.edit_mode}")
 
     def mousebutton_callback(self):
-        # print(self.mouse_position)
-        if self.widget_at_mouse != None:
-            self.reset_fields()
+        if self.widget_at_mouse is not None:
+            self.reset_highlighted_fields()
             self.highlight_fields(self.widget_at_mouse)
         else:
-            self.reset_fields()
-
+            self.reset_highlighted_fields()
 
     def entry_callback(self, widget):
-
-        entry_value = widget.entry_variable.get()  
-
+        entry_value = widget.entry_variable.get()
         if len(entry_value) > 1:
             widget.entry_variable.set(widget.entry_variable.get()[1:])
             entry_value = widget.entry_variable.get()
-
-                    
-        if (not entry_value.isdigit() or entry_value == "0"):
+        if not entry_value.isdigit() or entry_value == "0":
             widget.entry_variable.set("")
             entry_value = ''
-     
-
-        # self.invalid_field(widget.position[0], widget.position[1])
-        
-
-        
-
-
+            self.set_field_valid(widget.get_position()[0], widget.get_position()[1])
+        else:
+            entry_value = int(entry_value)
+            self.validate_field(widget)
+        if self.invalid_fields:
+            for row, column in self.invalid_fields:
```

Comment:
I think the problem is that when the list changes while running this loop the loop still continues with the old version of the list